File: ClientApp/src/app/models/OrdersDecorations.ts
import { Decoration } from "./Decoration";
import { Order } from "./Order";

export class OrdersDecorations {

    constructor(order?: Order, decoration?: Decoration, quantity?: number){
        if(order){
            this.order = order;
        }
        if(decoration){
            this.decoration_Id = decoration.decoration_Id;
            this.decoration = decoration;
        }
        this.quantity = quantity ? quantity : 1;
    }

    id: number;
    order_Id: number;
    decoration_Id: number;
    quantity: number;
    order: Order;
    decoration: Decoration;
    details?: string;
}

File: Models/OrdersDecorations.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Bakery.Models
{
    public class OrdersDecorations
    {
        [Key]
        public int Id { get; set; }
        public int Order_Id { get; set; }
        public int Decoration_Id { get; set; }
        public int Quantity { get; set; }
        [ForeignKey("Order_Id")]
        public virtual Order Order { get; set; }
        [ForeignKey("Decoration_Id")]
        public virtual Decoration Decoration { get; set; }
    }
}
